pkg/flux/flux.go

```diff
@@ -0,0 +1,169 @@
+package flux
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/weaveworks/weave-gitops/pkg/runner"
+	"github.com/weaveworks/weave-gitops/pkg/version"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . Flux
+type Flux interface {
+	CreateSourceGit(name string, url string, branch string, secretRef string, namespace string) ([]byte, error)
+	CreateSourceHelm(name string, url string, namespace string) ([]byte, error)
+	CreateKustomization(name string, source string, path string, namespace string) ([]byte, error)
+	CreateHelmReleaseGitRepository(name string, source string, path string, namespace string) ([]byte, error)
+	CreateHelmReleaseHelmRepository(name string, chart string, namespace string) ([]byte, error)
+	CreateSecretGit(name string, url string, namespace string) ([]byte, error)
+}
+
+type FluxClient struct {
+	runner runner.Runner
+}
+
+func New(cliRunner runner.Runner) *FluxClient {
+	return &FluxClient{
+		runner: cliRunner,
+	}
+}
+
+func (f *FluxClient) CreateSourceGit(name string, url string, branch string, secretRef string, namespace string) ([]byte, error) {
+	args := []string{
+		"create", "source", "git", name,
+		"--url", url,
+		"--branch", branch,
+		"--secret-ref", secretRef,
+		"--namespace", namespace,
+		"--interval", "30s",
+		"--export",
+	}
+
+	out, err := f.runFluxCmd(args...)
+	if err != nil {
+		return out, fmt.Errorf("failed to create source git: %w", err)
+	}
+
+	return out, nil
+}
+
+func (f *FluxClient) CreateSourceHelm(name string, url string, namespace string) ([]byte, error) {
+	args := []string{
+		"create", "source", "helm", name,
+		"--url", url,
+		"--namespace", namespace,
+		"--interval", "30s",
+		"--export",
+	}
+
+	out, err := f.runFluxCmd(args...)
+	if err != nil {
+		return out, fmt.Errorf("failed to create source helm: %w", err)
+	}
+
+	return out, nil
+}
+
+func (f *FluxClient) CreateKustomization(name string, source string, path string, namespace string) ([]byte, error) {
+	args := []string{
+		"create", "kustomization", name,
+		"--path", path,
+		"--source", source,
+		"--namespace", namespace,
+		"--prune", "true",
+		"--validation", "client",
+		"--interval", "1m",
+		"--export",
+	}
+
+	out, err := f.runFluxCmd(args...)
+	if err != nil {
+		return out, fmt.Errorf("failed to create kustomization: %w", err)
+	}
+
+	return out, nil
+}
+
+func (f *FluxClient) CreateHelmReleaseGitRepository(name string, source string, chartPath string, namespace string) ([]byte, error) {
+	args := []string{
+		"create", "helmrelease", name,
+		"--source", "GitRepository/" + source,
```

Comment:
In the "previous implementation", we needed to post-process this as flux was replacing ".wego" with "./wego". Perhaps that is no longer a problem? It's difficult to tell with such widespread change.

Reply:
thank you for catching that up, I remember seeing it but forgot to add it back.